### PR TITLE
deprecated api tri

### LINKED ISSUE #26865: [MNT]: Remove 3.7-deprecated API
# Summary

As old and unused API (classes, methods, functions) are found, they are marked deprecated with removal intended in a later release. These API are decorated or otherwise wrapped with [the deprecation functions in `_api`](https://github.com/matplotlib/matplotlib/blob/dceb278bef361f32eb0009dec5bfda41fc27fa38/lib/matplotlib/_api/__init__.py#L19-L22) (i.e., `deprecated`, `warn_deprecated`, `rename_parameter`, `delete_parameter`, `make_keyword_only`, `deprecate_method_override`, or `deprecate_privatize_attribute`).

Now that 3.8 is out and `main` targets 3.9, we can start the process of removing API that were deprecated in 3.7 (i.e., those with `since="3.7"` and without `pending=True`).

# Proposed fix

Each part of the deprecated API should be removed:
1. The deprecated code itself:
    a. Items marked with `_api.warn_deprecated`, `@_api.deprecated` or `@_api.deprecate_privatize_attribute` are to be removed on expiry, i.e., any of the marked lines below:
    ```diff
    -@api.deprecated("3.7", ...)
    -def foo():
    -    pass

     def bar(x):
         if x is None:
    -        _api.warn_deprecated("3.7", message="x=None is deprecated")
    +        raise ValueError("x=None is not supported")
        ...

     class Baz:
          ...
    -     value = _api.deprecate_privatize_attribute("3.7", ...)
    ```
    b. Items decorated with `@_api.rename_parameter` should have this decorator removed, and no other changes will be necessary (the parameter rename should already have been made when the decorator was added.), i.e.,
    ```diff
    -@_api.rename_parameter("3.7", "old_name", "new_name")
     def foo(new_name):
        ...
    ```
    c. Items decorated with `@_api.make_keyword_only` should have the decorator removed, and the parameter named there should be made keyword-only (by adding a `*` before that point).
    ```diff
    -@_api.make_keyword_only("3.7", "arg3")
    -def foo(arg1, arg2, arg3, arg4):
    +def foo(arg1, arg2, *, arg3, arg4):
        ...
    ```
    d. Items decorated with `@_api.delete_parameter` will need to have the parameter deleted _and_ all following parameters made keyword-only.
    ```diff
    -@_api.delete_parameter("3.7", "arg3")
    -def foo(arg1, arg2, arg3, arg4):
    +def foo(arg1, arg2, *, arg4):
        ...
    ```
3. The corresponding type stub in the related `*.pyi` file.
4. Any tests for _only_ deprecated behaviour.
5. Any related exceptions in `ci/mypy-stubtest-allowlist.txt`, if there are any.

Additionally, the removal should be documented by placing a snippet in `doc/api/next_api_changes/removals/` (see `00001-ABC.rst` in that directory for a template). For writing these snippets, you may wish to [copy the original deprecation notice](https://matplotlib.org/stable/api/prev_api_changes/api_changes_3.7.0.html#deprecations), modifying it slightly to state that the API has been removed instead. Please ensure that any references to deleted API include a full path (as there may be many methods with the same name, but on different classes.)

**NOTE**: After removing the deprecated API, there may be some additional fallout that should be corrected. For example,
1. Removed code may be aliased by importing in another location, so those imports will have to be removed.
2. Documentation may link to the removed API; these should be kept in code style. For example,
    ```rst
    This is some text referencing :class:`.RemovedClass`.
    ```
    should be changed to:
    ```rst
    This is some text referencing ``matplotlib.submodule.RemovedClass``.
    ```
    (Please expand the API into the full import name.)

See also [our documentation on removing deprecated API](https://matplotlib.org/devdocs/devel/contribute.html#expiring) and #26853 for an example removing several types of deprecated API.

# Task list
A quick search shows the following sets of possible parts to remove (for GHC OSD, please use Slack and/or Zoom to make sure someone else hasn't started working on the same set):
1. [x] `lib/matplotlib/animation.py` - 2 deprecations #26872
4. [x] `lib/matplotlib/axis.py` - 1 deprecation #26871
5. [x] `lib/matplotlib/backends/*.py` - 3 deprecations and `src/_backend_agg_wrapper.cpp` - 2 deprecations #26962
6. [x] `lib/matplotlib/cm.py` - 2 deprecations (Note: the documentation for this removal may be a bit tricky to write.) #26965
7. [x] `lib/matplotlib/contour.py` - 5 deprecations  #26907
8. [x] `lib/matplotlib/collections.py` - 2 deprecations #26874
9. [x] `lib/matplotlib/_fontconfig_patterns.py` - 1 deprecation #26884
10. [x] `lib/matplotlib/gridspec.py` - 1 deprecation  #26885 (!)
11. [x] `lib/matplotlib/lines.py` - 2 deprecations  #26902
12. [x] `lib/matplotlib/offsetbox.py` - 3 deprecations  #26910 ~~#26880~~
13. [x] `lib/matplotlib/patches.py` - 1 deprecation #26890 #26876 #26900 (!)
14. [x] `lib/matplotlib/quiver.py` - 2 deprecations #26918 
15. [x] `lib/matplotlib/tri/*.py` - 8 deprecations  #26909 
16. [x] `lib/mpl_toolkits/axisartist/axislines.py` - 2 deprecations #26900

There may be some other items I've missed here, but note that we do not want to remove any with `pending=True`, or `since="3.8"` or higher.

### PR DESCRIPTION
**Pull request by vigneshvetrivel8**
## PR summary
Helps with #26865 (lib/matplotlib/tri/*.py).
## PR checklist

- [ ] "closes part of #26865 (lib/matplotlib/tri/*.py)" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by ksunden** [comment]
The deprecation here is to remove the edited files entirely, as their contents/organization is now considered private (and thus they have `_tri<x>` python modules).

**Review by QuLogic — CHANGES_REQUESTED** [review]
This is not removing the deprecations, but changing imports in files that are deprecated, which does not achieve the desired result. In fact, it causes _more_ things to appear in the deprecated files.

**Review by ksunden — CHANGES_REQUESTED** [review]
Please do not edit the 00001-ABC.rst file, and instead make a new file e.g. `26909-VV.rst`

You can copy the contents (and update to reflect removal instead of deprecation) from, which gives more context and says what to do instead:

https://github.com/matplotlib/matplotlib/blob/d60de5ed4d374d688463a1a49684f6cd1dbdd0d5/doc/api/prev_api_changes/api_changes_3.7.0/deprecations.rst?plain=1#L81-L86

File: doc/api/next_api_changes/removals/26909-VV.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ ``matplotlib.tri`` submodules are deprecated 
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
+  
+ The ``matplotlib.tri.*`` submodules are deprecated.  All functionality is 
+ available in ``matplotlib.tri`` directly and should be imported from there.
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/26909-VV.rst:5]: The indent makes it a code block, and doesn't get rendered as a heading:
```suggestion
``matplotlib.tri`` submodules are removed
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The ``matplotlib.tri.*`` submodules are removed.  All functionality is
available in ``matplotlib.tri`` directly and should be imported from there.
```

**Comment by vigneshvetrivel8** [comment]
Do i need to modify some more or is this enough. Is this PR ok.

**Review by QuLogic — APPROVED** [review]
This looks good, except for the pre-commit linting issue. I'm going to have it fix it automatically for you.

**Comment by QuLogic** [comment]
pre-commit.ci autofix

**Review by ksunden — APPROVED** [review]

**Comment by QuLogic** [comment]
Thanks @vigneshvetrivel8! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Pull request merged**